### PR TITLE
BUGFIX: Fix regression breaking login to Neos

### LINKED ISSUE #2860: Login to Neos broken with 5.0.6 and 5.1.2
### Description

Going to `/neos` all you get is an error, status code 500.

`Exception in line 382 of /…/Neos_Neos_Controller_LoginController.php: Call to undefined method Neos\Flow\Mvc\ActionResponse::setHeader()`

Yes, broken through my upmerges… see https://github.com/neos/neos-development-collection/commit/4f67bf10be335e6321f6c83fb818cddee16e59e0#diff-6163b96fe6022b470a5505ab6cad0ee1R116

### Affected Versions

Neos: 5.0.6, 5.1.2

### PR DESCRIPTION
**Pull request by kdambekalns**
Fixes #2860

### CODE REVIEW

**Comment by kdambekalns** [comment]
@kitsunet Yes, `true` was cast to `'1'` before. But do we really need to do that manually now?

**Comment by kitsunet** [comment]
Yes, we do, this is low level HTTP interaction, you better know what you are doing. ;)

**Comment by kdambekalns** [comment]
Well, `MessageTrait` does cast to string in `trimHeaderValues()`…

File: Neos.Neos/Classes/Controller/LoginController.php [hunk 1]
```diff
@@ -112,8 +113,8 @@ public function initializeIndexAction()
      */
     public function indexAction($username = null, $unauthorized = false)
     {
-        if ($this->securityContext->getInterceptedRequest() || $unauthorized) {
-            $this->response->setHeader('X-Authentication-Required', true);
+        if ($unauthorized || $this->securityContext->getInterceptedRequest()) {
```
Comment (kdambekalns) [inline Neos.Neos/Classes/Controller/LoginController.php]: Note: this is just re-ordering, putting the cheaper operation first. 😎

**Comment by kitsunet** [comment]
True, but according to spec the setHeader() method accepts strings or arrays of strings, so I would go by that for now.

**Pull request merged**